File: src/visualization/visualizer.cpp
#include "visualizer.h"
#include "../utils/node.h"

void drawGrid(sf::RenderWindow& window, Grid& grid, int size) {
    for (int i = 0; i < grid.rows; ++i) {
        for (int j = 0; j < grid.cols; ++j) {
            grid.getNode(i, j)->draw(window, size);
        }
    }
}

File: src/visualization/visualizer.py
import pygame

WHITE, GREY = (255, 255, 255), (128, 128, 128)

def draw_grid(screen, grid, width, rows):
    screen.fill(WHITE)
    for row in grid:
        for node in row:
            node.draw(screen)
    for i in range(rows):
        pygame.draw.line(screen, GREY, (0, i * width // rows), (width, i * width // rows))
        pygame.draw.line(screen, GREY, (i * width // rows, 0), (i * width // rows, width))
    pygame.display.flip()
